aviatrix/resource_aviatrix_transit_gateway.go

```diff
@@ -871,13 +914,33 @@ func resourceAviatrixTransitGatewayCreate(d *schema.ResourceData, meta interface
 		gateway.Interfaces = b64.StdEncoding.EncodeToString(interfaceList)
 
 		if goaviatrix.IsCloudType(cloudType, goaviatrix.EDGENEO) {
-			// set the static interface mapping for AEP
-			interfaceMapping := map[string][]string{
-				"eth0": {"wan", "0"},
-				"eth1": {"wan", "1"},
-				"eth2": {"wan", "2"},
-				"eth3": {"mgmt", "0"},
-				"eth4": {"wan", "3"},
+			/*
+				TODO: Add support for user provided interface mapping. Currently, the interface mapping is hardcoded for Dell devices. This will be updated in a future release.
+			*/
+			interfaceMapping := map[string][]string{}
```

Comment:
So interface mappings will only ever be required for edge is that right?  And is that the long term plan or we plan to remove interface mapping altogether in the future?

Reply:
The plan is to remove interface_mapping in the future and internally create a mapping in the code using the device_id provided by the user.